app/webpacker/components/wca/RegionSelector.jsx

```diff
@@ -1,54 +1,85 @@
 
 export const ALL_REGIONS_VALUE = 'all';
 
+const allRegionsOption = { key: 'all', text: I18n.t('common.all_regions'), value: ALL_REGIONS_VALUE };
+
+const continentOptions = Object.values(continents.real).map((continent) => (
+  { key: continent.id, text: continent.name, value: continent.id }
+));
+
+const countryOptions = Object.values(countries.real).map((country) => (
+  {
+    key: country.id,
+    text: country.name,
+    value: country.iso2,
+    flag: (
+      <>
+        <RegionFlag iso2={country.iso2} withoutTooltip />
+        {' '}
+      </>
+    ),
+  }
+));
+
 const regionsOptions = [
-  { key: 'all', text: I18n.t('common.all_regions'), value: ALL_REGIONS_VALUE },
+  allRegionsOption,
   {
     key: 'continents_header',
     value: '',
     disabled: true,
     content: <Header content={I18n.t('common.continent')} size="small" style={{ textAlign: 'center' }} />,
   },
-  ...(Object.values(continents.real).map((continent) => (
-    { key: continent.id, text: continent.name, value: continent.id }
-  ))),
+  ...continentOptions,
   {
     key: 'countries_header',
     value: '',
     disabled: true,
     content: <Header content={I18n.t('common.country')} size="small" style={{ textAlign: 'center' }} />,
   },
-  ...(Object.values(countries.real).map((country) => (
-    {
-      key: country.id,
-      text: country.name,
-      value: country.iso2,
-      flag: { className: country.iso2.toLowerCase() },
-    }
-  ))),
+  ...countryOptions,
 ];
 
-export default function RegionSelector({ region, onRegionChange, clearable = false }) {
-  const defaultValue = clearable ? null : ALL_REGIONS_VALUE;
+/**
+ * For `region`, pass `ALL_REGIONS_VALUE`, a continent id, or a country iso2.
+ *
+ * To omit non-country options, use `onlyCountries`.
+ */
+export default function RegionSelector({
+  onlyCountries = false,
+  label = I18n.t(
+    onlyCountries ? 'common.country' : 'competitions.index.region',
+  ),
+  region,
+  onRegionChange,
+  nullable = false,
+  disabled = false,
+  error = null,
+  name,
+}) {
+  const defaultValue = (nullable || onlyCountries) ? null : ALL_REGIONS_VALUE;
 
   return (
-    <>
-      <label htmlFor="region">{I18n.t('competitions.index.region')}</label>
-      <Dropdown
-        search
-        selection
-        clearable={region !== defaultValue}
-        value={region}
-        options={regionsOptions}
-        // clearing calls onChange with the empty string; catch and replace it
-        onChange={(_, data) => onRegionChange(data.value || defaultValue)}
-      />
-    </>
+    <Form.Select
+      label={label}
+      name={name}
+      search
+      selection
+      clearable={
+        (nullable && region !== null) || (defaultValue && region !== defaultValue)
+      }
+      value={region}
+      options={onlyCountries ? countryOptions : regionsOptions}
+      // clearing calls onChange with the empty string; catch and replace it
+      onChange={(_, data) => onRegionChange(data.value || defaultValue)}
```

Comment:
I think for the sake of reusability, you should not ignore the event and instead propagate the "long form" of the setter (the one reacting to `(evt, { value })`'s `value`. Outside code that used to have a "plain" setter can then "wrap" this SemUI-esque function signature when needed.

See also how `useInputState` achieves a similar result -- it effectively returns a setter function that can be called in two different ways.

Reply:
That makes sense :+1: